test/e2e/SelectNetworkAndAccount.test.ts

```diff
@@ -33,64 +35,118 @@ describe('connectAndFetchAccounts', () => {
   };
 
+  const mockWeb3Enable: () => Promise<InjectedExtension[]> = vi.fn().mockResolvedValue([
+    {
+      name: 'polkadot-js',
+      version: '0.50.0',
+      injector: {
+        signer: {} as any,
+        accounts: {
+          get: vi.fn(async () => [
+            { address: '5FHneW46xGXgs5mUiveU4sbTyGBzmto6UXo7B6a', meta: { name: 'Alice' } } as InjectedAccount,
+          ]),
+          subscribe: vi.fn((cb: (accounts: InjectedAccount[]) => void) => {
+            cb([{ address: '5FHneW46xGXgs5mUiveU4sbTyGBzmto6UXo7B6a', meta: { name: 'Alice' } } as InjectedAccount]);
+            return () => {};
+          }),
+        },
+      },
+    } as unknown as InjectedExtension,
+  ]);
+
+  const mockWeb3EnableEmptyAccounts: () => Promise<InjectedExtension[]> = vi.fn().mockResolvedValue([
+    {
+      name: 'polkadot-js',
+      version: '0.50.0',
+      injector: {
+        signer: {} as any,
+        accounts: {
+          get: vi.fn(async () => []),
+          subscribe: vi.fn((cb: (accounts: InjectedAccount[]) => void) => {
+            cb([]);
+            return () => {};
+          }),
+        },
+      },
+    } as unknown as InjectedExtension,
+  ]);
+
+  const mockWeb3Accounts = vi.fn().mockResolvedValue([
+    {
+      address: '5FHneW46xGXgs5mUiveU4sbTyGBzmto6UXo7B6a',
+      meta: { name: 'Alice' },
+      type: 'sr25519',
+    } as InjectedAccountWithMeta,
+    {
+      address: '5DAAnrj7VHTz5s9Uo29F9u1ZKx4u3ZtX5kh6p',
+      meta: { name: 'Bob' },
+      type: 'ed25519',
+    } as InjectedAccountWithMeta,
+  ]);
+
+  const mockWeb3EmptyAccounts = vi.fn().mockResolvedValue([]);
+
   beforeEach(() => {
     vi.clearAllMocks();
     allAccountsStore.set(new Map([[mockAccount.address, mockAccount]]));
     user.set({ ...mockAccount });
   });
 
   it('sets networkErrorMsg when endpoint is missing', async () => {
-    const { container, component } = render(SelectNetworkAndAccount, {
+    render(SelectNetworkAndAccount, {
       props: {
         newUser: null,
         accounts: new Map(),
       },
     });
 
     // invalid endpoint
-    await component.connectAndFetchAccounts({ name: 'testnet', endpoint: null });
-
     expect(
-      getByText(container, 'Could not connect to empty value. Please enter a valid and reachable Websocket URL.')
-    ).toBeDefined();
+      connectAndFetchAccounts(
+        { id: NetworkType.TESTNET_PASEO, name: 'Testnet', pathName: 'testnet', endpoint: undefined },
+        mockWeb3Enable,
+        mockWeb3Accounts
+      )
+    ).rejects.toThrowError('Could not connect to empty value. Please enter a valid and reachable Websocket URL.');
   });
 
-  it('sets successfully calls createApi and fetchAccountsForNetwork', async () => {
-    const disconnectMock = vi.fn();
-    (createApi as unknown as Mock).mockResolvedValue({
-      api: { disconnect: disconnectMock },
-    });
+  // it('sets successfully calls createApi and fetchAccountsForNetwork', async () => {
```

Comment:
nit: remove commented-out tests, or if keeping test to fix later, un-comment and mark as `it.skip(...)`